sites/partners/src/listings/PaperListingForm/index.tsx

```diff
@@ -456,54 +442,87 @@ const ListingForm = ({ listing, editMode }: ListingFormProps) => {
-        const formattedData = formatFormData(
-          data,
-          units,
-          openHouseEvents,
-          orderedPreferences,
-          latLong,
-          customMapPositionChosen
-        )
-        const result = editMode
-          ? await listingsService.update({
-              listingId: listing.id,
-              body: { id: listing.id, ...formattedData },
-            })
-          : await listingsService.create({ body: formattedData })
-        setLoading(false)
-        if (result) {
-          setSiteAlertMessage(
-            editMode ? t("listings.listingUpdated") : t("listings.listingSubmitted"),
-            "success"
+    async (formData: FormListing) => {
+      if (!loading) {
+        try {
+          setLoading(true)
+          clearErrors()
+          const orderedPreferences = preferences.map((pref, index) => {
+            return { ...pref, ordinal: index + 1 }
+          })
+          const formattedData = formatFormData(
+            formData,
+            units,
+            openHouseEvents,
+            orderedPreferences,
+            latLong,
+            customMapPositionChosen
           )
-
-          await router.push(`/listings/${result.id}`)
+          const removeEmptyFields = (obj) => {
+            Object.keys(obj).forEach(function (key) {
+              if (obj[key] && typeof obj[key] === "object") {
+                removeEmptyFields(obj[key])
+              }
+              if (obj[key] === null || obj[key] === undefined || obj[key] === "") {
+                delete obj[key]
+              }
+              if (
+                typeof obj[key] === "object" &&
+                !Array.isArray(obj[key]) &&
+                Object.keys(obj[key]).length === 0
+              ) {
+                delete obj[key]
+              }
+            })
+          }
+          removeEmptyFields(formattedData)
+
+          const result = editMode
+            ? await listingsService.update({
+                listingId: listing.id,
+                body: { id: listing.id, ...formattedData },
+              })
+            : await listingsService.create({ body: formattedData })
+          reset(formData)
+          if (result) {
+            setSiteAlertMessage(
+              editMode ? t("listings.listingUpdated") : t("listings.listingSubmitted"),
+              "success"
+            )
+
+            await router.push(`/listings/${result.id}`)
+          }
+          setLoading(false)
+        } catch (err) {
+          reset(formData)
+          setLoading(false)
+          setAlert("api")
+          const { data } = err.response || {}
+          if (data.statusCode === 400) {
+            data?.message?.forEach((errorMessage: string) => {
+              const fieldName = errorMessage.split(" ")[0]
+              const readableError = getReadableErrorMessage(errorMessage)
+              if (readableError) {
+                setError(fieldName, { message: readableError })
+              }
+            })
```

Comment:
Shouldn't we catch other errors (different than 400) as well?

Reply:
We are catching all errors but showing field-specific errors on 400